pkg/resource/deploy/step_generator_test.go

```diff
@@ -360,3 +360,105 @@ func TestEngineDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestResourceInheritsOptionsFromParent(t *testing.T) {
```

Comment:
The three subtests in this package might be easily combinable into a table test.

```go
tests := []struct{
    desc string

    parentDeletedWith string
    childDeletedWith  string
    wantDeletedWith   string
}{
    {
        desc: "...",
        parentDeletedWith: "parent-deleted-with",
        childDeletedWith: "",
        wantDeletedWith: "parent-deleted-with",
    },
    {
        desc: "...",
        parentDeletedWith: "parent-deleted-with",
        childDeletedWith: "this-value-should-not-change",
        wantDeletedWith: "this-value-should-not-change",
    },
    {
        desc: "...",
        parentDeletedWith: "",
        childDeletedWith: "this-value-should-not-change",
        wantDeletedWith: "this-value-should-not-change",
    },
}

for _, tt := range tests {
    tt := tt
    t.Run(tt.desc, func(t *testing.T) {
        t.Parallel()

        parentURN := // ...
        parentGoal := &resource.Goal{
            // ...
            DeletedWith: tt.parentDeletedWith,
        }

        childURN := // ...
        goal := &resource.Goal{
            // ...
            DeletedWith: tt.childDeletedWith,
        }

        sg := // ...
        // ...

        require.NoError(t, err)
        assert.Equal(t, tt.wantDeletedWith, goal.DeletedWith)
    })
}
```

You can keep the test general (as it is) to leave room for other options here,
or maybe rename it to TestInheritsDeletedWithFromParent to make it
DeletedWith-specific, which will also allow renaming the fields of the table
above to something shorter:

```go

func TestInheritsDeletedWithFromParent(t *testing.T) {
    t.Parallel()

    tests := []struct{
        desc   string
        parent string
        child  string
        want   string
    }{
        // ...
```